### ISSUE TITLE
Oni2 only interpreting multi-key binds if keys are pressed simultaneously

### ISSUE THREAD

**Issue by lukamanitta** [onivim/oni2#3125]
**OS**
macOS Big Sur v11.1
**Oni2 Version**
0.5.9-nightly (0.5.1870)

This is the contents of my keybindings config file:
```
[  
      { "key": "jk", "command": "vim.esc", "when": "insertMode" },
      { "key": "<Leader>e", "command": "workbench.view.explorer", "when": "!insertMode && !textInputFocus" }
]
```

When I try to use, for example, the "<Leader>e" keybind to switch to/from the explorer window, nothing happens unless I press both keys (my leader is set to space with "vim.leader": "<Space>") at the exact same time; only then is the appropriate function executed.

Up until a few days ago (apologies, I don't remember if there was an update / what version it was), both of these keybinds worked properly; when I pressed space, or "j" in insert mode, the editor would wait until I pressed a subsequent key before executing the function of the first. 

I do have my config folder symlinked from a separate dotfiles folder, but I have tested this issue without the symlink and it persists.

I've tried changing my key repeat speed and delay, but neither change had any effect.

I've had a look at the input debug file, and when I press, for example, "Space", it interprets the key for one "frame" or "tick" only.

I'm not sure where to start with troubleshooting this any further, apologies for not being able to provide any more information.

Edit: a reinstall fixes the problem (for now), but I'm still confused as to why this happened in the first place.

**Comment by bryphe** [comment]
Hi @lukamanitta ,

Thanks for reporting the issue.

I suspect this is related to https://github.com/onivim/oni2/pull/3102 - but I can't reproduce. In that PR, there's a default timeout of 1 second set for input. After 1s, the keys get flushed. 

It sounds like somehow, though, the timeout is being treated as instant.

We could see if this is related by setting the `vim.timeout` variable in your `configuration.json`:
- `"vim.timeout": 0` should turn off the timeout
- `"vim.timeout": 2500` would change the timeout to `2.5s`

Actually just saw your last edit - I'm glad it's working now!

If you see it again, we could look in the log file to see (via `oni2 -f -log-file=input.log`) - specifically looking for the delta between the `keydown`, `text`, `keyup` sequence, and the `Timeout` action:
```
[INFO]  [ 7.432s] Feature.Input.ReveryKeyConverter : Key info: Unmodified: j
 WithShift: J
 WithAltGr: ∆
 WithAltGrShift: Ô


[INFO]  [ 7.432s] Oni2.Store.dispatch : dispatch: KeyDown {
  key =
  [(KeyPress.PhysicalKey
      { PhysicalKey.key = (Key.Character <opaque>);
        modifiers =
        { Modifiers.control = false; alt = false; altGr = false;
          shift = false; super = false }
        })
    ];
  scancode = 13; time = <opaque>}
[INFO]  [ 7.433s] Oni2.Store.dispatch : dispatch: (TextInput ("j", <opaque>))
[INFO]  [ 7.435s] Oni2.Store.dispatch : dispatch: (Notification AnimateYOffset {id = 1; msg = <opaque>})
[INFO]  [ 7.448s] Oni2.Store.dispatch : dispatch: (Input Feature_Input.Timeout)
[INFO]  [ 7.449s] Oni2.Store.dispatch : dispatch: KeyTimeout
```

**Comment by bryphe** [comment]
Ah, I think I know what's going on - https://github.com/onivim/oni2/pull/3102 had a bug in second-to-millisecond conversion - which was fixed here: https://github.com/onivim/oni2/pull/3104  So the default configuration would say that the timeout should be 1 second, but when we read it back (...or process the default setting), it'd be interpreted as 1 millisecond. 

But the nightly build didn't have that fix (it must've just missed the cut-off)... Explains why re-installing addressed it, at least. Sorry you hit this in between that fix!

I'll close this since this it seems this was fixed by #3104 - but please let me know if you still see any quirks around this.

**Comment by lukamanitta** [comment]
Ah, great. Thanks for the info! Oni2 looks like it's going to be my main IDE from now on, and the great dev communication and speed makes me even more excited! Thanks @bryphe :)
